tests/expressions/tests.py

```diff
@@ -149,6 +149,80 @@ def test_update(self):
             ],
         )
 
+    def test_slicing_in_f_expressions(self):
+        # F expressions using string-fields can be sliced.
+        tests = [
+            (F('name')[:], 'Example Inc.', 'Example Inc.'),
+            (F('name')[:7], 'Example Inc.', 'Example'),
+            (F('name')[0], 'Example', 'E'),
+            (F('name')[5], 'E', ''),
+            (F('name')[7:], 'Foobar Ltd.', 'Ltd.'),
+            (F('name')[0:10], 'Ltd.', 'Ltd.'),
+            (F('name')[2:7], 'Test GmbH', 'st Gm'),
+        ]
+        for expression, name, expected in tests:
+            with self.subTest(expression=expression, name=name, expected=expected):
+                company = Company.objects.get(name=name)
+                company.name = expression
+                company.save()
+                company.refresh_from_db()
+                self.assertEqual(company.name, expected)
+
+    def test_slicing_in_f_expressions_with_annotate(self):
+        companies = Company.objects.annotate(first_three=F('name')[:3])
+        self.assertSequenceEqual(
+            companies.values_list('first_three', flat=True),
+            ['Exa', 'Foo', 'Tes']
+        )
+        companies = Company.objects.annotate(after_three=F('name')[3:])
+        self.assertSequenceEqual(
+            companies.values_list('after_three', flat=True),
+            ['mple Inc.', 'bar Ltd.', 't GmbH']
+        )
+        companies = Company.objects.annotate(random_four=F('name')[2:5])
+        self.assertSequenceEqual(
+            companies.values_list('random_four', flat=True),
+            ['amp', 'oba', 'st ']
+        )
+        companies = Company.objects.annotate(first_letter=F('name')[:1])
+        self.assertSequenceEqual(
+            companies.values_list('first_letter', flat=True),
+            ['E', 'F', 'T']
+        )
+        companies = Company.objects.annotate(first_letter=F('name')[0])
+        self.assertSequenceEqual(
+            companies.values_list('first_letter', flat=True),
+            ['E', 'F', 'T']
+        )
+
+    def test_negative_index_in_slicing_f_expressions(self):
+        msg = "Negative indexing is not supported."
+        with self.assertRaisesMessage(ValueError, msg):
+            company = Company.objects.filter(name="Foobar Ltd.")
+            company.update(name=F('name')[:-4])
```

Comment:
Please convert this to use `self.subTest()` and add a third case for negative `slice.start`.